static/panes/editor.js

```diff
@@ -267,67 +267,76 @@ Editor.prototype.initCallbacks = function () {
 };
 
 Editor.prototype.initButtons = function (state) {
-    this.fontScale = new FontScale(this.domRoot, state, this.editor);
-    this.languageBtn = this.domRoot.find('.change-language');
+    var scopedThis = this;
+
+    scopedThis.fontScale = new FontScale(scopedThis.domRoot, state, scopedThis.editor);
+    scopedThis.languageBtn = scopedThis.domRoot.find('.change-language');
     // Ensure that the button is disabled if we don't have nothing to select
     // Note that is might be disabled for other reasons beforehand
-    if (this.langKeys.length <= 1) {
-        this.languageBtn.prop("disabled", true);
+    if (scopedThis.langKeys.length <= 1) {
+        scopedThis.languageBtn.prop("disabled", true);
     }
-    var addCompilerButton = this.domRoot.find('.btn.add-compiler');
-    var paneAdderDropdown = this.domRoot.find('.add-pane');
+    scopedThis.topBar = scopedThis.domRoot.find('.top-bar');
+    scopedThis.hideable = scopedThis.domRoot.find('.hideable');
+
+    scopedThis.loadSaveButton = scopedThis.domRoot.find('.load-save');
+    var paneAdderDropdown = scopedThis.domRoot.find('.add-pane');
+    var addCompilerButton = scopedThis.domRoot.find('.btn.add-compiler');
+    scopedThis.conformanceViewerButton = scopedThis.domRoot.find('.btn.conformance');
+    var addEditorButton = scopedThis.domRoot.find('.btn.add-editor');
 
     var togglePaneAdder = function () {
         paneAdderDropdown.dropdown('toggle');
     };
-    this.topBar = this.domRoot.find('.top-bar');
 
     // NB a new compilerConfig needs to be created every time; else the state is shared
     // between all compilers created this way. That leads to some nasty-to-find state
     // bugs e.g. https://github.com/mattgodbolt/compiler-explorer/issues/225
     var getCompilerConfig = _.bind(function () {
-        return Components.getCompiler(this.id, this.currentLanguage.id);
+        return Components.getCompiler(scopedThis.id, scopedThis.currentLanguage.id);
```

Comment:
scopedThis and this are the same thing here. This is a bit confusing to read.

Reply:
Yes they are the same thing for this underscore bind function. The confusion comes from other parts where `this` and `scopedThis` are not the same.